src/test/java/integration/IntegrationTest.java

```diff
@@ -3,563 +3,1428 @@
+        req = new FileChangeRequest(file1.getFileID(), changes, file1.getFileVersion()).getRequest(response -> {
+            Assert.assertNotEquals("Authenticated method succeeded with no auth info", 200, response.getStatus());
 
             waiter.release();
         }, errHandler);
-        wsMgr.sendRequest(req);
+
+        ws3.sendRequest(req);
         if (!waiter.tryAcquire(5, TimeUnit.SECONDS)) {
             Assert.fail("Acquire timed out");
         }
-    }
-
-    private void testUserLogin() throws InterruptedException, ConnectException {
-        logger.info(String.format("Logging in"));
-        Semaphore waiter = new Semaphore(0);
 
-        req = new UserLoginRequest(user1ID, user1Pass).getRequest(response -> {
-            Assert.assertEquals("Failed to log in", 200, response.getStatus());
 
-            sender1Token = ((UserLoginResponse) response.getData()).getToken();
-            wsMgr.setAuthInfo(user1ID, sender1Token);
-            sender1ID = user1ID;
+        /*
+         * Test invalid fileID
+         */
+        req = new FileChangeRequest(-1, changes, -1).getRequest(response -> {
+            Assert.assertNotEquals("Failed to throw error on invalid file", 200, response.getStatus());
 
             waiter.release();
         }, errHandler);
-        wsMgr.sendRequest(req);
+
+        ws1.sendRequest(req);
         if (!waiter.tryAcquire(5, TimeUnit.SECONDS)) {
             Assert.fail("Acquire timed out");
         }
-    }
 
-    private void testProjectGetPermissionConstants() throws InterruptedException {
-        logger.info("Requesting api permission constants");
-        Semaphore waiter = new Semaphore(0);
+        /*
+         * Test invalid permissions
+         */
+        req = new FileChangeRequest(file2.getFileID(), changes, file2.getFileVersion()).getRequest(response -> {
+            Assert.assertNotEquals("Failed to throw permissions error on lack of write permissions", 200, response.getStatus());
 
-        req = new ProjectGetPermissionConstantsRequest().getRequest( response -> {
-            Assert.assertEquals("Failed to get permission constants", 200, response.getStatus());
-            apiConstants = ((ProjectGetPermissionConstantsResponse) response.getData()).getConstants();
             waiter.release();
         }, errHandler);
 
-        wsMgr.sendRequest(req);
+        ws1.sendRequest(req);
         if (!waiter.tryAcquire(5, TimeUnit.SECONDS)) {
             Assert.fail("Acquire timed out");
         }
 
-        Assert.assertNotEquals("Constants map empty", 0, apiConstants.size());
-
-        // NOTE: I chose a permission which (in my opinion) will probably always be around
-        //       if the "read" permission is ever removed from the server, this will fail - Joel
-        Assert.assertTrue("Constants map does not contain correct provlages", apiConstants.containsKey("read"));
+        /*
+         * Test file version too high
+         */
     }
 
-    private void testProjec
```

Comment:
Error message here should probably say something about the path not being the same, not just changes.